modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/artifact/DeployableArtifact.java

```diff
@@ -49,4 +57,15 @@
    * @return MuleContext created from the artifact configurations files.
    */
   MuleContext getMuleContext();
+
+  /**
+   * @return the directory where the artifact content is stored.
+   */
+  File getLocation();
+
+  /**
+   * @return a service to test connection over configuration components.
+   */
+  ConnectivityTestingService getConnectivityTestingService();
```

Comment:
I don't like having to expose the connectivity service here just because it is required on one use case. I think that will end up exposing many things as in the muleContext. Maybe a more generic way to obtain services by class can help